src/snapshot.rs

```diff
@@ -0,0 +1,57 @@
+#![allow(dead_code)]
+use std::{fs, io};
+
+use ethers::types::U256;
+use indexmap::IndexSet;
+use serde::{Deserialize, Serialize};
+
+use crate::GENESIS_BLOCK;
+
+/// Struct containing the fields used for restoring the tree state.
+#[derive(Serialize, Deserialize)]
+pub struct StateSnapshot {
```

Comment:
Idea for the future: Could we store the latest root hash here as well and use that to verify that tree & index key map are in sync?

Reply:
Great idea!